### ISSUE TITLE
Trainer.py module 'datasets' has no attribute 'Dataset'

### ISSUE THREAD

**Issue by FrancoisMentec** [huggingface/transformers#7337]
I'm trying to use a Trainer, but I get this error:
```
c:\users\francois\appdata\local\programs\python\python37\lib\site-packages\transformers\trainer.py in __init__(self, model, args, data_collator, train_dataset, eval_dataset, tokenizer, model_init, compute_metrics, tb_writer, optimizers, **kwargs)
    287 
    288         if is_datasets_available():
--> 289             if isinstance(train_dataset, datasets.Dataset):
    290                 self._remove_unused_columns(self.train_dataset, description="training")
    291             if isinstance(eval_dataset, datasets.Dataset):

AttributeError: module 'datasets' has no attribute 'Dataset'
```
My guess is that `datasets.Dataset` should be replaced by `torch.utils.data.Dataset` but I haven't checked the source file. Maybe the person responsible for the `Trainer` development should look into that.

I'm using transformers version 3.2.0 btw.

**Comment by sgugger** [comment]
I'm not sure which `datasets` module is installed in your env, but the Hugging Face `datasets` definitely has a `Dataset` attribute. And no, this part is not using PyTorch `Dataset`.

**Comment by FrancoisMentec** [comment]
Apparently, the `datasets` module wasn't even installed on my environment. But installing it just replaced the error by another one. It's upgrading PyTorch that fixed the issue, might be cool to be notified during the installation or the execution of transformers that we don't have the required PyTorch version for it to works. It feels awkward having to fixe dependencies myself, that's what a package manager like pip is usually used for. Maybe it's better handled by anaconda...

**Comment by ericdoug-qi** [comment]
hi, could you please tell me which pytorch version you have been upgraded to solve this problem? I got the same problem.

**Comment by ericdoug-qi** [comment]
> Apparently, the `datasets` module wasn't even installed on my environment. But installing it just replaced the error by another one. It's upgrading PyTorch that fixed the issue, might be cool to be notified during the installation or the execution of transformers that we don't have the required PyTorch version for it to works. It feels awkward having to fixe dependencies myself, that's what a package manager like pip is usually used for. Maybe it's better handled by anaconda...

hi, could you please tell me which pytorch version you have been upgraded to solve this problem? I got the same problem.

**Comment by FrancoisMentec** [comment]
@ericdoug-qi Can't remember, did you check you are using the latest version of PyTorch and Transformers? Otherwise, try anaconda.

**Comment by joejoeqiiq** [comment]
may be you use the wrong moudule, try to run conmand "pip uninstall datasets" , then it can use  Hugging Face datasets

**Comment by m-chaves** [comment]
This solved it for me:
I checked the dependencies of `datasets`.
`pip show datasets` 
I uninstalled all dependencies and the `datasets` package. 
`pip uninstall aiohttp dill filelock fsspec huggingface-hub multiprocess numpy packaging pandas pyarrow pyarrow-hotfix pyyaml requests tqdm xxhash datasets`
Then I reinstalled `datasets`, which also installs the dependencies as well but with the proper versions that do not conflict. 
`pip install datasets`
Finally I restarted the kernel.
